File: backend/src/models/Notification.js
const { DataTypes } = require('sequelize');
const { sequelize } = require('../config/database');
const Utilisateur = require('./utilisateur');
const Transaction = require('./Transaction');

const Notification = sequelize.define('Notification', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true,
    allowNull: false,
  },
  titre: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  message: {
    type: DataTypes.TEXT,
    allowNull: false,
  },
  type_notification: {
    type: DataTypes.ENUM('commande_recue', 'commande_confirmee', 'paiement_requis', 'transaction_validee'),
    allowNull: false,
  },
  est_lu: {
    type: DataTypes.BOOLEAN,
    defaultValue: false,
    allowNull: false,
  },
  id_utilisateur: {
    type: DataTypes.INTEGER,
    allowNull: false,
    references: { model: Utilisateur, key: 'id' }
  },
  id_transaction: {
    type: DataTypes.INTEGER,
    allowNull: true, // Nullable car pas toutes les notifs sont liées à une transaction
    references: { model: Transaction, key: 'id' }
  },
}, {
  timestamps: true,
});

// Relations
Notification.belongsTo(Utilisateur, { foreignKey: 'id_utilisateur', as: 'utilisateur' });
Utilisateur.hasMany(Notification, { foreignKey: 'id_utilisateur', as: 'notifications' });

Notification.belongsTo(Transaction, { foreignKey: 'id_transaction', as: 'transaction' });
Transaction.hasMany(Notification, { foreignKey: 'id_transaction', as: 'notifications' });

module.exports = Notification;

File: frontend/lib/models/Notification.dart
class NotificationModel {
  final int id;
  final String titre;
  final String message;
  final String typeNotification;
  final bool estLu;
  final int idUtilisateur;
  final int? idTransaction;
  final DateTime createdAt;
  final Map<String, dynamic>? transaction;

  NotificationModel({
    required this.id,
    required this.titre,
    required this.message,
    required this.typeNotification,
    required this.estLu,
    required this.idUtilisateur,
    this.idTransaction,
    required this.createdAt,
    this.transaction,
  });

  factory NotificationModel.fromJson(Map<String, dynamic> json) {
    return NotificationModel(
      id: json['id'],
      titre: json['titre'],
      message: json['message'],
      typeNotification: json['type_notification'],
      estLu: json['est_lu'],
      idUtilisateur: json['id_utilisateur'],
      idTransaction: json['id_transaction'],
      createdAt: DateTime.parse(json['createdAt']),
      transaction: json['transaction'],
    );
  }
}